source/Engine.cpp

```diff
@@ -2046,7 +2046,8 @@ void Engine::DoCollection(Flotsam &flotsam)
 	for(Body *body : shipCollisions.Circle(flotsam.Position(), 5.))
 	{
 		Ship *ship = reinterpret_cast<Ship *>(body);
-		if(!ship->CannotAct() && ship != flotsam.Source() && ship->Cargo().Free() >= flotsam.UnitSize())
+		if(!ship->CannotAct() && ship != flotsam.Source() && ship->Cargo().Free() >= flotsam.UnitSize() &&
+			ship->GetGovernment() != flotsam.SourceGovernment())
```

Comment:
Might be considered a nitpick, but I'd shuffle the order of operations here to check if the governments don't match before checking if you have enough free space. (`can I act AND is this not mine AND is this not an ally's AND do I have space` instead of `can I act AND is this not mine AND do I have space AND is this not an ally's`.)
```suggestion
		if(!ship->CannotAct() && ship != flotsam.Source() && ship->GetGovernment() != flotsam.SourceGovernment()
			&& ship->Cargo().Free() >= flotsam.UnitSize()
```